### ISSUE TITLE
Recognize `Nullable<T>(T value)` when inferring nullability of `Value` field

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#32626]
No warning should be reported for `y.Value.F`:
```C#
using System;

struct S
{
    internal object? F;
}

class Program
{
    static void F()
    {
        S? x = new Nullable<S>(new S());
        x.Value.F.ToString(); // warning

        S? y = new Nullable<S>(new S() { F = 2 });
        y.Value.F.ToString(); // ok
    }
}
```

[jcouv:] As part of this we should test a nullable tuple: `(string, string)? t = (null, ""); ...`

**Comment by cston** [comment]
See also https://github.com/dotnet/roslyn/issues/32531.
